src/nativestore/PropertyLink.cpp

```diff
@@ -15,33 +15,50 @@ limitations under the License.
 
 #include <sstream>
 #include <vector>
+#include <iostream>
 
-#include "../../util/logger/Logger.h"
+#include "../util/logger/Logger.h"
 
 Logger property_link_logger;
+unsigned int PropertyLink::nextPropertyIndex = 1;  // Starting with 1 because of the 0 and '\0' differentiation issue
+//std::string PropertyLink::DB_PATH = "/home/ubuntu/software/jasminegraph/streamStore/properties.db";
+std::fstream* PropertyLink::propertiesDB = NULL;
+pthread_mutex_t lockPropertyLink;
+pthread_mutex_t lockCreatePropertyLink;
+pthread_mutex_t lockInsertPropertyLink;
+pthread_mutex_t lockGetPropertyLink;
+
+
+
+
 
 PropertyLink::PropertyLink(unsigned int propertyBlockAddress) : blockAddress(propertyBlockAddress) {
+    //The problem is when we create the PropertyLink.
+    pthread_mutex_lock(&lockPropertyLink);
     if (propertyBlockAddress > 0) {
-        this->propertiesDB->seekg(propertyBlockAddress);
+        PropertyLink::propertiesDB->seekg(propertyBlockAddress * PropertyLink::PROPERTY_BLOCK_SIZE);
         char rawName[PropertyLink::MAX_NAME_SIZE] = {0};
+//        property_link_logger.info("Traverse state  = " + std::to_string(PropertyLink::propertiesDB->rdstate()));
 
         if (!this->propertiesDB->read(reinterpret_cast<char*>(&rawName), PropertyLink::MAX_NAME_SIZE)) {
-            property_link_logger.error("Error while reading property name from block " + std::to_string(blockAddress));
+            property_link_logger.log("XXXXXXXX : PropertyLink", "info");
```

Comment:
Delete this line,

`property_link_logger.log("XXXXXXXX : PropertyLink", "info");`

Reply:
Done.

Reply:
Ok